### PR TITLE
The line chart cannot move to the left

### LINKED ISSUE #143: The line chart can't move to the left
Hi, lightweight-charts team
I am facing an issue with lightweight-charts The steps to reproduce the issue are :

- I create chart in div LightweightCharts.createChart
- I using addLineSeries and set data 3 times

when scroll to position left in the chart, the chart can't move to left. and show error message in console` Uncaught Error: Value is null`

**Code**
```html
<!DOCTYPE html>
<html>

<head>
    <meta charset='utf-8'>
    <meta http-equiv='X-UA-Compatible' content='IE=edge'>
    <meta name='viewport' content='width=device-width, initial-scale=1'>
    <script src='https://unpkg.com/lightweight-charts/dist/lightweight-charts.standalone.production.js'></script>
</head>

<body>
    <div id="chart"></div>
    <script>
        const contentChart = document.getElementById('chart');
        const chart = LightweightCharts.createChart(contentChart, { width: 1200, height: 500 });
        const line1 = chart.addLineSeries();
        line1.setData([
            { time: '2017-04-11', value: 80.01 },
            { time: '2017-04-12', value: 96.63 },
            { time: '2017-04-13', value: 76.64 },
            { time: '2017-04-14', value: 81.89 },
            { time: '2017-04-15', value: 74.43 },
            { time: '2017-04-16', value: 80.01 },
            { time: '2017-04-17', value: 96.63 },
            { time: '2017-04-18', value: 76.64 },
            { time: '2017-04-19', value: 81.89 },
            { time: '2017-04-20', value: 81.89 },
            { time: '2017-04-21', value: 81.89 },
            { time: '2017-04-22', value: 81.89 },
            { time: '2017-04-23', value: 81.89 }
        ]);

        const line2 = chart.addLineSeries();
        line2.setData([
            { time: '2018-04-11', value: 80.01 },
            { time: '2018-04-12', value: 96.63 },
            { time: '2018-04-13', value: 76.64 },
            { time: '2018-04-14', value: 81.89 },
            { time: '2018-04-15', value: 74.43 },
            { time: '2018-04-16', value: 80.01 },
            { time: '2018-04-17', value: 96.63 },
            { time: '2018-04-18', value: 76.64 },
            { time: '2018-04-19', value: 81.89 },
            { time: '2018-04-20', value: 81.89 },
            { time: '2018-04-21', value: 81.89 },
            { time: '2018-04-22', value: 81.89 },
            { time: '2018-04-23', value: 81.89 }
        ]);

        const line3 = chart.addLineSeries();
        line3.setData([
            { time: '2019-04-11', value: 80.01 },
            { time: '2019-04-12', value: 96.63 },
            { time: '2019-04-13', value: 76.64 },
            { time: '2019-04-14', value: 81.89 },
            { time: '2019-04-15', value: 74.43 },
            { time: '2019-04-16', value: 80.01 },
            { time: '2019-04-17', value: 96.63 },
            { time: '2019-04-18', value: 76.64 },
            { time: '2019-04-19', value: 81.89 },
            { time: '2019-04-20', value: 81.89 },
            { time: '2019-04-21', value: 81.89 },
            { time: '2019-04-22', value: 81.89 },
            { time: '2019-04-23', value: 81.89 }
        ]);
    </script>
</body>

</html>
```

**Lightweight Charts Version:** 1.0.2

**Actual behavior:**
The line chart can't move to the left

**Expected behavior:**
The line chart can move to the left

**Screenshots**
I record event on scroll in chart
https://drive.google.com/file/d/1iRsv5VMmbEeC1l4zmYLSo6IyJiWkMQf8/view

### PR DESCRIPTION
**Pull request by kirchet**
**Type of PR:** bugfix

**PR checklist:**

- [x] Addresses an existing issue: fixes #143 
- [x] Includes tests
- [ ] Documentation update

**Overview of change:**

Getting first value from other series in case when the main series of the pane is out of the viewport

### CODE REVIEW

File: src/model/chart-model.ts [hunk 1]
```diff
@@ -373,7 +373,12 @@ export class ChartModel implements IDestroyable {
 	public setAndSaveCurrentPosition(x: Coordinate, y: Coordinate, pane: Pane): void {
 		this._crosshair.saveOriginCoord(x, y);
 		let price = NaN;
-		const index = this._timeScale.coordinateToIndex(x);
+		let index = this._timeScale.coordinateToIndex(x);
+
+		const visibleBars = this._timeScale.visibleBars();
+		if (visibleBars !== null) {
+			index = Math.min(Math.max(visibleBars.firstBar(), index), visibleBars.lastBar()) as TimePointIndex;
```
Comment (timocov) [inline src/model/chart-model.ts]: How coordinate might be outside of the viewport?

File: src/model/price-scale.ts [hunk 2]
```diff
@@ -541,6 +541,26 @@ export class PriceScale {
...
+		let result: FirstValue | null = null;
+
+		for (const source of this._dataSources) {
+			if (source instanceof PriceDataSource) {
+				const firstValue = source.firstValue();
+				if (firstValue === null) {
+					continue;
+				}
+
+				if (result === null || result.timePoint > firstValue.timePoint) {
```
Comment (timocov) [inline src/model/price-scale.ts]: What do you think about swapping the condition to the following?
```suggestion
				if (result === null || firstValue.timePoint < result.timePoint) {
```

File: src/model/chart-model.ts (hunk 1 above)
Comment (kirchet) [reply src/model/chart-model.ts]: It can't, but the aligned (nearest) index can be outside of the viewport

File: src/model/price-scale.ts (hunk 2 above)
Comment (kirchet) [reply src/model/price-scale.ts]: Looks good

File: src/model/chart-model.ts (hunk 1 above)
Comment (timocov) [reply src/model/chart-model.ts]: Ah, sure!

**Pull request merged**